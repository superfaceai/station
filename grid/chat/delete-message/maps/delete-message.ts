/* eslint-disable jest/no-export */

import { RecordingProcessOptions, SuperfaceTest } from '@superfaceai/testing';

export const deleteMessageTest = (
  provider: string,
  destination: string[],
  invalidMessageId: string,
  options?: RecordingProcessOptions
): void => {
  describe(`chat/delete-message/${provider}`, () => {
    let superface: SuperfaceTest;
    let prepare: SuperfaceTest;

    describe('DeleteMessage', () => {
      beforeAll(() => {
        superface = new SuperfaceTest({
          profile: 'chat/delete-message',
          provider,
          useCase: 'DeleteMessage',
          testInstance: expect,
        });

        prepare = new SuperfaceTest({
          profile: 'chat/send-message',
          useCase: 'SendMessage',
          provider,
        });
      });

      describe('when specified destination does exist', () => {
        let messageId: string;

        beforeEach(async () => {
          const result = await prepare.run({
            input: {
              destination: destination[0],
              text: 'test DeleteMessage',
            },
            testName: 'prepare-chat/delete-chat/send',
          });

          if (result.isOk()) {
            // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access, @typescript-eslint/no-unsafe-assignment
            messageId = (result.value as any).messageId;
          }
        });

        it('performs correctly', async () => {
          const result = await superface.run(
            {
              profile: 'chat/delete-message',
              useCase: 'DeleteMessage',
              input: {
                destination: destination[0],
                messageId,
              },
            },
            { ...options, hideInput: ['messageId'] }
          );

          expect(result.isOk).toBeTruthy();
          expect(result).toMatchSnapshot();
        });
      });

      describe('when specified destination does not exist', () => {
        it('returns error', async () => {
          const result = await superface.run(
            {
              input: {
                destination: destination[1],
                messageId: invalidMessageId,
              },
            },
            options
          );

          expect(result).toMatchSnapshot();
        });
      });

      describe('when specified message id does not exist', () => {
        it('returns error', async () => {
          const result = await superface.run(
            {
              input: {
                destination: destination[0],
                messageId: invalidMessageId,
              },
            },
            options
          );

          expect(result).toMatchSnapshot();
        });
      });
    });
  });
};
