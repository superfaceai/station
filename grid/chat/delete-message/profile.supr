"""
Delete a message
Delete a text message from channel destination
"""
name = "chat/delete-message"
version = "1.0.0"

"""
Delete Message
Deletes a message from one destination
"""
usecase DeleteMessage {
  input {
    """
    Destination
    Channel to delete the message from
    """
    destination! string!
    
    """
    Message Id
    Id of message to delete
    """
    messageId! string!
  }

  result {
    """
    Rate Limit
    Rate limit details.
    """
    rateLimit RateLimit
  }

  error Error

  example Successful {
    input {
      destination = 'C1H9RESGL',
      messageId = '1503435956.000247',
    }

    result {
      success = true,
    }
  }

  example Failed {
    input {
      destination = 'random',
      messageId = '1503435956.000247',
    }

    error {
      title = 'invalid destination ID',
    }
  }
}

model Error {
  """
  Title
  A short, human-readable summary of the problem type.
  """
  title!

  """
  Detail
  A human-readable explanation specific to this occurrence of the problem.
  """
  detail

  """
  Rate Limit
  Rate limit details.
  """
  rateLimit RateLimit
}

model RateLimit {
  """
  Bucket
  Different parts of API may have different rate limits.
  Bucket identifies to which part of API the rate limits apply.
  """
  bucket string

  """
  Total Requests
  Total requests available in the time window.
  """
  totalRequests number

  """
  Remaining Requests
  Remaining requests available in the time window.
  """
  remainingRequests number

  """
  Remaining Requests available in percents
  Remaining requests available in the time window in percents.
  """
  remainingRequestsPercentage number

  """
  Reset Timestamp
  Timestamp when the rate limits will reset (in Unix Timestamp ms format).
  """
  resetTimestamp number

  """
  Reset After
  Number of seconds until reset of rate limit
  """
  resetAfter number

  """
  Retry After
  Number of seconds until user can retry the request
  """
  retryAfter number
}
