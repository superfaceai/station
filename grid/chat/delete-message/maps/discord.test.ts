import { deleteMessageTest } from './delete-message';

/**
 * Live tests
 *
 * @group live/safe
 */

deleteMessageTest(
  'discord',
  ['938614740995960844', '000000000000000000'],
  '000000000000000000'
);
