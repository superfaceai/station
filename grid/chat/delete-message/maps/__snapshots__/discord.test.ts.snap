// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`chat/delete-message/discord DeleteMessage when specified destination does exist performs correctly 1`] = `
Ok {
  "value": Object {
    "rateLimit": Object {
      "bucket": "da101873bd9ceb21138638aa25f33cb1",
      "remainingRequests": 4,
      "remainingRequestsPercentage": 80,
      "resetAfter": 5,
      "resetTimestamp": 1664267734000,
      "totalRequests": 5,
    },
  },
}
`;

exports[`chat/delete-message/discord DeleteMessage when specified destination does not exist returns error 1`] = `
Err {
  "error": "HTTPError: Expected HTTP error
Properties: {
  \\"title\\": \\"Not found\\",
  \\"detail\\": \\"Unknown Channel\\",
  \\"rateLimit\\": {
    \\"bucket\\": \\"da101873bd9ceb21138638aa25f33cb1\\",
    \\"totalRequests\\": 5,
    \\"remainingRequests\\": 4,
    \\"resetAfter\\": 5,
    \\"resetTimestamp\\": 1664267734000,
    \\"remainingRequestsPercentage\\": 80
  }
}
AST Path: definitions[0].statements[2].responseHandlers[1].statements[1]
Original Map Location: Line 36, column 7",
}
`;

exports[`chat/delete-message/discord DeleteMessage when specified message id does not exist returns error 1`] = `
Err {
  "error": "HTTPError: Expected HTTP error
Properties: {
  \\"title\\": \\"Not found\\",
  \\"detail\\": \\"Unknown Message\\",
  \\"rateLimit\\": {
    \\"bucket\\": \\"da101873bd9ceb21138638aa25f33cb1\\",
    \\"totalRequests\\": 5,
    \\"remainingRequests\\": 3,
    \\"resetAfter\\": 5,
    \\"resetTimestamp\\": 1664267734000,
    \\"remainingRequestsPercentage\\": 60
  }
}
AST Path: definitions[0].statements[2].responseHandlers[1].statements[1]
Original Map Location: Line 36, column 7",
}
`;
